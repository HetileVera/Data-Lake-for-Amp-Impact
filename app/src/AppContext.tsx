import React, { createContext, useContext } from "react";
import isEmpty from "lodash/isEmpty";
import throttle from "lodash/throttle";

import { AppStore, IAppStore } from "./models/AppStore";
import { IInstallation } from "./models/Installation";
import { isLocalDev, websiteUrlOrigin, assetWebsiteUrl, secretKey } from "./helpers/settings";
import { onSnapshot } from "mobx-state-tree";

// Create an instance of the root store (a.k.a appStore)
const appStore = AppStore.create({});

// Store state to local storage
const storeLocally = throttle(() => appStore.saveLocal(), 5000);

// Store state in S3 when applicable
const storeRemotely = throttle(() => appStore.saveRemote(), 30000);

// The initialization function for the root store
export function initializeStore() {
  if (!isEmpty(secretKey)) console.warn("Access key is bundled!");

  if (isLocalDev) {
    console.log(`isLocalDev '${isLocalDev}'`);
    console.log(`websiteUrlOrigin '${websiteUrlOrigin}'`);
    console.log(`assetWebsiteUrl '${assetWebsiteUrl}'`);

    if (window.location.host !== "localhost:3000") {
      console.warn("You should run the app on localhost!");

      throw new Error("App should run on localhost for development mode!");
    }
  }

  appStore.loadLocal();
  onSnapshot(appStore, (snapshot) => {
    storeLocally();
    storeRemotely();
  });
}

// Create the store context
export const StoreContext = createContext<IAppStore | undefined>(undefined);

// Create the store provider component
export const StoreProvider: React.FunctionComponent = ({ children }) => {
  return <StoreContext.Provider value={appStore}>{children}</StoreContext.Provider>;
};

// useStore hook
export function useStore(): IAppStore {
  const store = useContext(StoreContext);

  if (!store) {
    throw new Error("useStore() must be used within a StoreProvider");
  }

  return store;
}

// useInstallation hook
export function useInstallation(): IInstallation {
  const store = useStore();
  const installation = store.installation;

  if (!installation) {
    throw new Error("An installation instance was not created yet");
  }

  return installation;
}
