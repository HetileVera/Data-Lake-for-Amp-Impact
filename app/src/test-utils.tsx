import React from "react";
import { render, RenderOptions } from "@testing-library/react";
import { ChakraProvider } from "@chakra-ui/react";

import { theme } from "themes/orange";

export const CustomChakraProvider = ({ children }: { children: React.ReactNode }) => {
  return (
    <ChakraProvider theme={theme}>
      <div data-testid="test-root">{children}</div>
    </ChakraProvider>
  );
};

const customRender = (ui: React.ReactElement, options?: Omit<RenderOptions, "wrapper">) => render(ui, { wrapper: CustomChakraProvider, ...options });

export * from "@testing-library/react";

export { customRender as render };
