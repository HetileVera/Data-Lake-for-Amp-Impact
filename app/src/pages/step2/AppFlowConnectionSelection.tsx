import React from "react";
import isEmpty from "lodash/isEmpty";
import { Box, Heading, HStack, Button, Select, Progress, Alert, AlertIcon, IconButton } from "@chakra-ui/react";
import { RepeatIcon } from "@chakra-ui/icons";
import { observer } from "mobx-react";

import { useInstallation } from "AppContext";
import { useConnectionsStore } from "models/ConnectionsStore";

export const AppFlowConnectionSelection = observer(() => {
  const installation = useInstallation();
  const { isError, isLoading, isReloading, store } = useConnectionsStore();
  const connectionNames = store.connectionNames;
  const connectionName = installation.appFlowConnectionName;

  const handleChange: React.ChangeEventHandler<HTMLSelectElement> = (event) => {
    installation.setAppFlowConnectionName(event.target.value);
  };

  const handleTryAgain = async () => {
    store.load();
  };

  React.useEffect(() => {
    if (!isEmpty(connectionNames) && connectionNames?.length === 1) {
      // If we only have one connection name, we will use it
      installation.setAppFlowConnectionName(connectionNames[0]);
    } else if (isEmpty(connectionNames) && !isLoading && !isReloading) {
      installation.setAppFlowConnectionName("");
    }
  }, [connectionNames, installation, isLoading, isReloading]);

  React.useEffect(() => {
    if (isError) {
      // When there is an error, we want to clear any selection
      installation.setAppFlowConnectionName("");
    }
  }, [isError, installation]);

  return (
    (isLoading && (
      <Box mt={6} mb={8}>
        <Box textAlign="center" fontWeight="bold" color="blue.500" fontSize="sm" mb={2}>
          Retrieving AppFlow Connection Names
        </Box>
        <Progress id="appflow-connection-loader" size="sm" isIndeterminate colorScheme="blue" bg="blue.75" />
      </Box>
    )) ||
    (isError && !isReloading && (
      <Alert status="error" variant="left-accent" mt={0} mb={8} color="red.700" alignItems="flex-end">
        <AlertIcon alignSelf="flex-start" />
        <Box>
          <Box>
            Something went wrong and we are unable to connect to your AWS account to get the list of Salesforce connections. This might be an
            intermittent problem. Wait for a few minutes and try again.
          </Box>
          <Box textAlign="right" w="full" mt={4}>
            <Button colorScheme="red" size="sm" onClick={handleTryAgain} loadingText="Processing" isLoading={isLoading || isReloading}>
              Try Again
            </Button>
          </Box>
        </Box>
      </Alert>
    )) ||
    (isEmpty(connectionNames) && (
      <Alert status="info" variant="left-accent" mt={0} mb={8} color="blue.700" alignItems="flex-end">
        <AlertIcon alignSelf="flex-start" />
        <Box>
          <Box>
            We are unable to find the connection name. It might take a few minutes before the connection name is available. Wait for a few seconds and
            try again.
          </Box>
          <Box textAlign="right" w="full" mt={4}>
            <Button colorScheme="blue" size="sm" onClick={handleTryAgain} loadingText="Processing" isLoading={isLoading || isReloading}>
              Try Again
            </Button>
          </Box>
        </Box>
      </Alert>
    )) || (
      <Box bg="blue.75" p={6} mb={8} borderRadius="lg" border="1px" borderColor="blue.100">
        <Heading id="step2-h2-select-connection" size="sm" pb="10px" color="blue.600">
          Select the connection name
        </Heading>

        <HStack>
          <Select bg="blue.50" id="appFlowConnectionName" value={connectionName} onChange={handleChange} disabled={isLoading || isReloading}>
            <option value="">Select a connection name</option>
            {connectionNames?.map((name, index) => (
              <option value={name} key={index}>
                {name}
              </option>
            ))}
          </Select>
          <IconButton
            id="step2-button-reload-connections"
            colorScheme="blue"
            variant="outline"
            aria-label="Search database"
            onClick={handleTryAgain}
            icon={<RepeatIcon />}
            isLoading={isLoading || isReloading}
          />
        </HStack>
      </Box>
    )
  );
});
