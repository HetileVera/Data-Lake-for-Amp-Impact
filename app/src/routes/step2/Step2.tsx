import React from "react";
import isEmpty from "lodash/isEmpty";
import {
  Box,
  Container,
  Heading,
  Text,
  HStack,
  Button,
  Link as ChakraLink,
  ChakraProvider,
  UnorderedList,
  ListItem,
  Stack,
  Collapse,
  Divider,
} from "@chakra-ui/react";
import { ArrowForwardIcon, ArrowBackIcon, ExternalLinkIcon } from "@chakra-ui/icons";
import { observer } from "mobx-react";
import { useNavigate } from "react-router-dom";

import appflowConnectionImage01 from "images/appflow-connection-01.png";
import appflowConnectionImage02 from "images/appflow-connection-02.png";
import appflowConnectionImage03 from "images/appflow-connection-03.png";
import appflowConnectionImage04 from "images/appflow-connection-04.png";

import { useInstallation, useStore } from "AppContext";
import { theme } from "themes/blue";
import { CurvedBox } from "components/CurvedBox";
import { StepsIndicator } from "components/StepsIndicator";
import { ClickableImage } from "components/ClickableImage";
import { YesNoAnswer } from "models/steps/BaseStep";
import { OutlineButton } from "components/OutlineButton";
import { Header } from "components/Header";
import { AppFlowConnectionSelection } from "routes/step2/AppFlowConnectionSelection";

export const Step2 = observer(() => {
  const installation = useInstallation();
  const region = installation.region;
  const step = installation.connectToSalesforceStep;
  const createdConnection = step.createdConnection;
  const navigate = useNavigate();
  const [isSubmitting, setSubmitting] = React.useState(false);
  const appStore = useStore();

  const handleAnswer = (answer: YesNoAnswer) => () => {
    step.setCreatedConnection(answer);
  };

  const handlePrevious = () => {
    navigate("/steps/1");
  };

  const handleSubmit: React.MouseEventHandler<HTMLButtonElement> = async () => {
    if (isEmpty(installation.appFlowConnectionName)) throw Error("No appFlow connection name is selected");
    step.markCompleted();
    setSubmitting(false);
    navigate("/steps/3");
  };

  React.useEffect(() => {
    step.markStarted();
    window.scrollTo(0, 0);
  }, [appStore, step]);

  const canProceed = createdConnection === YesNoAnswer.Yes && !isEmpty(installation.appFlowConnectionName);

  return (
    <ChakraProvider theme={theme}>
      <CurvedBox bgGradient={theme.gradients.bgLight} />

      <Header />

      <Box position="relative">
        <CurvedBox />

        <Box bg="blue.600" w="full" position="relative">
          <Container maxW="container.md" pt="15px" pb="15px">
            <StepsIndicator current={2} />
          </Container>
        </Box>

        <Container maxW="container.md" pt="0px" position="relative">
          <Heading display="inline-block" size="lg" pt="16px" pb="30px" color="gray.100" letterSpacing="-1px">
            Connect to your Salesforce organization
          </Heading>

          <Box color="gray.50" mb={10}>
            <Stack direction="column" spacing="8px" align="left" justifyContent="space-between">
              <Box>For AWS to access your Salesforce data, we will need your authorization.</Box>
              <Box>We will guide you through the steps of setting up Amazon AppFlow which is a service that allows AWS to connect to Salesforce.</Box>
              <Box>You will be creating a Salesforce connection, signing in to Salesforce and authorizing access.</Box>
            </Stack>
          </Box>

          <Box borderRadius="lg" boxShadow="base" bg="blue.50" mt={4} p={8}>
            <Box p={0} mb={6}>
              <Heading size="md" pt="0px" pb="10px" color="blue.600" letterSpacing="-1px">
                Create a dedicated Salesforce user
              </Heading>
              <Text mb={4}>
                To gain access to your Salesforce data, you need a Salesforce user for authentication purposes. It is best security practice to use a
                dedicated user for this purpose. This helps ensure business continuity and adheres to the principle of least privilege to help secure
                your data. For detailed instructions, see &nbsp;
                <ChakraLink href="https://help.salesforce.com/articleView?id=000331470&amp;type=1&amp;mode=1" isExternal color="blue.600">
                  Create a secure Salesforce API user <ExternalLinkIcon mx="2px" />
                </ChakraLink>
              </Text>
              If you do not have an extra full Salesforce user license available, you can skip this step and instead repurpose a user with a profile
              or permission set that provides:
              <UnorderedList ml={5} mt={2} mb={3}>
                <ListItem>Access to the objects and fields in Salesforce that you plan to sync</ListItem>
                <ListItem>System Permission: API Enabled</ListItem>
                <ListItem> System Permission: Manage Connected Apps</ListItem>
              </UnorderedList>
              This is likely a user with a System Administrator or similar profile.
            </Box>

            <Box p={0}>
              <Heading size="md" pt="0px" pb="10px" color="blue.600" letterSpacing="-1px">
                Create a Salesforce connection using Amazon AppFlow
              </Heading>
              <Text mb={4}>
                Amazon AppFlow is a service that allows AWS to connect to Salesforce, for this to work, you need to create a Salesforce connection
                using Amazon AppFlow.
              </Text>
            </Box>

            <Stack direction={{ base: "column", md: "row" }} spacing="30px" pt={3} pb={6}>
              <Box>
                <ClickableImage w={{ base: "full", md: "300px" }} src={appflowConnectionImage01} title="AppFlow Connections Page" />
              </Box>
              <Box>
                <UnorderedList ml={5} mt={2} mb={3} color="blue.800">
                  <ListItem>
                    Go to the &nbsp;
                    <ChakraLink
                      href={`https://${region}.console.aws.amazon.com/appflow/home?region=${region}#/connections/salesforce`}
                      isExternal
                      color="blue.600"
                    >
                      AppFlow connections page <ExternalLinkIcon mx="2px" />
                    </ChakraLink>
                  </ListItem>
                  <ListItem> Click on the Create connection button </ListItem>
                </UnorderedList>
              </Box>
            </Stack>

            <Divider borderColor="blue.100" />

            <Stack direction={{ base: "column", md: "row" }} spacing="30px" pt={3} pb={6} mt={2}>
              <Box>
                <ClickableImage w={{ base: "full", md: "300px" }} src={appflowConnectionImage02} title="AppFlow Connection Page" />
              </Box>
              <Box>
                <UnorderedList ml={5} mt={2} mb={3} color="blue.800">
                  <ListItem>Choose the Salesforce environment you want to connect to</ListItem>
                  <ListItem>Leave the Private Link selection as 'Disabled'</ListItem>
                  <ListItem>Leave the Data Encryption setting as is</ListItem>
                  <ListItem>Type a name for the connection, it can be anything you want, then click Continue</ListItem>
                </UnorderedList>
              </Box>
            </Stack>

            <Divider borderColor="blue.100" />

            <Stack direction={{ base: "column", md: "row" }} spacing="30px" pt={3} pb={6} mt={3}>
              <Box>
                <ClickableImage w={{ base: "full", md: "300px" }} src={appflowConnectionImage03} title="Salesforce Login Page" />
              </Box>
              <Box>
                <UnorderedList ml={5} mt={2} mb={3} color="blue.800">
                  <ListItem>Provide the user name and password for the dedicated Salesforce user that you created earlier</ListItem>
                </UnorderedList>

                <Box fontSize="sm" mt={3} bg="blue.75" p={3} borderRadius="lg">
                  Remember to log in using the newly created dedicated Salesforce user
                </Box>
              </Box>
            </Stack>

            <Divider borderColor="blue.100" />

            <Stack direction={{ base: "column", md: "row" }} spacing="30px" pt={3} pb={6} mt={3}>
              <Box>
                <ClickableImage w={{ base: "full", md: "300px" }} src={appflowConnectionImage04} title="Salesforce Allow Access Page" />
              </Box>
              <Box>
                <UnorderedList ml={5} mt={2} mb={3} color="blue.800">
                  <ListItem>Allow access to your Salesforce data</ListItem>
                  <ListItem>
                    You will be directed back to the AppFlow connections page, you should see the connection name that you just created. You don't
                    need to do anything else on the AppFlow connections page. Go ahead and select Yes for the next question.
                  </ListItem>
                </UnorderedList>
              </Box>
            </Stack>

            <Stack direction="row" spacing="0px" pt={3} pb={6} alignItems="baseline" justifyContent="space-between">
              <Heading size="md" color="blue.600" letterSpacing="-1px" mt={0}>
                Did you create a Salesforce connection using Amazon AppFlow?
              </Heading>
              <Box textAlign="right">
                <OutlineButton selected={createdConnection === YesNoAnswer.Yes} onClick={handleAnswer(YesNoAnswer.Yes)} ml={3}>
                  Yes
                </OutlineButton>
              </Box>
            </Stack>

            <Collapse in={createdConnection === YesNoAnswer.Yes} animateOpacity>
              <Box>{createdConnection === YesNoAnswer.Yes && <AppFlowConnectionSelection />}</Box>
            </Collapse>
          </Box>

          <HStack justifyContent="space-between" p={3} pt={6} mb={12}>
            <Button colorScheme="blue" size="md" leftIcon={<ArrowBackIcon />} disabled={isSubmitting} variant="ghost" onClick={handlePrevious}>
              Previous
            </Button>
            <Button
              colorScheme="blue"
              size="md"
              loadingText="Processing"
              rightIcon={<ArrowForwardIcon />}
              ml={3}
              isLoading={isSubmitting}
              disabled={!canProceed || isSubmitting}
              onClick={handleSubmit}
            >
              Next
            </Button>
          </HStack>
        </Container>
      </Box>
    </ChakraProvider>
  );
});
