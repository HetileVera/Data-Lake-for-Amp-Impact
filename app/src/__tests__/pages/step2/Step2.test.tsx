import { cleanup, screen } from "@testing-library/react";
import { create } from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

import { render, RenderWithChakra } from "test-utils";

import * as appContext from "AppContext";
import { Step2 } from "pages/step2/Step2";

const spyScrollTo = jest.fn();

describe("Step2 component", () => {
  beforeEach(() => {
    Object.defineProperty(global.window, "scrollTo", { value: spyScrollTo });
  });

  afterEach(() => {
    spyScrollTo.mockClear();
    cleanup;
  });

  test("snapshot", () => {
    // @ts-ignore
    jest.spyOn(appContext, "useInstallation").mockImplementation(() => {
      return {
        id: "1",
        startDate: "2022-01-01",
        startedBy: "test-user",
        accountId: "test-account-id",
        region: "us-east-1",
        appFlowConnectionName: "test-appflow-connection-name",
        credentials: [],
        deploymentStep: "1",
        reviewStep: "1",
        nextStepNumber: "2",
        connectToSalesforceStep: {
          createdConnection: "Yes",
          markStarted: jest.fn(),
          markCompleted: jest.fn(),
          setCreatedConnection: jest.fn(),
        },
      };
    });

    const tree = create(
      <RenderWithChakra>
        <appContext.StoreProvider>
          <BrowserRouter>
            <Step2 />
          </BrowserRouter>
        </appContext.StoreProvider>
      </RenderWithChakra>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("render properly", () => {
    // @ts-ignore
    jest.spyOn(appContext, "useInstallation").mockImplementation(() => {
      return {
        id: "1",
        startDate: "2022-01-01",
        startedBy: "test-user",
        accountId: "test-account-id",
        region: "us-east-1",
        appFlowConnectionName: "test-appflow-connection-name",
        credentials: [],
        deploymentStep: "1",
        reviewStep: "1",
        nextStepNumber: "2",
        connectToSalesforceStep: {
          createdConnection: "Yes",
          markStarted: jest.fn(),
          markCompleted: jest.fn(),
          setCreatedConnection: jest.fn(),
        },
      };
    });

    render(
      <appContext.StoreProvider>
        <BrowserRouter>
          <Step2 />
        </BrowserRouter>
      </appContext.StoreProvider>
    );

    expect(screen.getAllByText(/connect to your salesforce organization/i)).toBeTruthy();
    expect(screen.getByText(/create a dedicated salesforce user/i)).toBeInTheDocument();
    expect(screen.getByText(/create a dedicated salesforce user/i)).toBeInTheDocument();
    expect(screen.getByText("Previous")).toBeInTheDocument();
    expect(screen.getByText("Next")).toBeInTheDocument();
  });
});
