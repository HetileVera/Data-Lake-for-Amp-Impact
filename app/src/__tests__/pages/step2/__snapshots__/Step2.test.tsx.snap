// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step2 component snapshot 1`] = `
Array [
  <div
    className="css-1v55i00"
    data-testid="curved-box"
  />,
  <div
    className="chakra-container css-1szj4qk"
  >
    <div
      className="chakra-stack css-olw0lh"
    >
      <div
        className="chakra-stack css-18lyn6q"
      >
        <div
          className="css-1i3fafl"
        >
          <img
            className="chakra-image css-0"
            src="Salesforce_Corporate_Logo_RGB.png"
          />
        </div>
        <div
          className="chakra-stack__divider css-r2vkku"
        />
        <div
          className="css-nutmff"
        >
          <img
            className="chakra-image css-ir5tdx"
            src="AWS_logo_RGB_BLK.png"
          />
        </div>
      </div>
      <div
        className="chakra-stack css-1n7gsjr"
      >
        <div
          className="css-18n4g2t"
        >
          <h2
            className="chakra-heading css-1opsxio"
          >
            Datalake for Nonprofits, Powered by AWS
          </h2>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="css-79elbk"
  >
    <div
      className="css-t7j6xp"
      data-testid="curved-box"
    />
    <div
      className="css-1hbx7hm"
    >
      <div
        className="chakra-container css-h66qep"
      >
        <div
          className="chakra-stack css-84zodg"
        >
          <span
            className="css-9xsoz7"
          >
            1
          </span>
          <span
            className="css-hfqsm"
          >
            2
          </span>
          <span
            className="css-ulmpab"
          >
            3
          </span>
          <span
            className="css-ulmpab"
          >
            4
          </span>
          <span
            className="css-ulmpab"
          >
            5
          </span>
          <span
            className="css-ulmpab"
          >
            6
          </span>
        </div>
      </div>
    </div>
    <div
      className="chakra-container css-11yz6e2"
    >
      <h2
        className="chakra-heading css-h8t1vd"
      >
        Connect to your Salesforce organization
      </h2>
      <div
        className="css-1cm40bb"
      >
        <div
          className="chakra-stack css-1wy5s38"
        >
          <div
            className="css-0"
          >
            For AWS to access your Salesforce data, we will need your authorization.
          </div>
          <div
            className="css-0"
          >
            We will guide you through the steps of setting up Amazon AppFlow which is a service that allows AWS to connect to Salesforce.
          </div>
          <div
            className="css-0"
          >
            You will be creating a Salesforce connection, signing in to Salesforce and authorizing access.
          </div>
        </div>
      </div>
      <div
        className="css-10a6eoe"
      >
        <div
          className="css-1frymol"
        >
          <h2
            className="chakra-heading css-1yxyor"
          >
            Create a dedicated Salesforce user
          </h2>
          <p
            className="chakra-text css-rszk63"
          >
            To gain access to your Salesforce data, you need a Salesforce user for authentication purposes. It is best security practice to use a dedicated user for this purpose. This helps ensure business continuity and adheres to the principle of least privilege to help secure your data. For detailed instructions, see  
            <a
              className="chakra-link css-oga2ct"
              href="https://help.salesforce.com/articleView?id=000331470&type=1&mode=1"
              rel="noopener noreferrer"
              target="_blank"
            >
              Create a secure Salesforce API user 
              <svg
                className="chakra-icon css-2y6psj"
                focusable={false}
                viewBox="0 0 24 24"
              >
                <g
                  fill="none"
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeWidth="2"
                >
                  <path
                    d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
                  />
                  <path
                    d="M15 3h6v6"
                  />
                  <path
                    d="M10 14L21 3"
                  />
                </g>
              </svg>
            </a>
          </p>
          If you do not have an extra full Salesforce user license available, you can skip this step and instead repurpose a user with a profile or permission set that provides:
          <ul
            className="css-12af3b8"
            role="list"
          >
            <li
              className="css-0"
            >
              Access to the objects and fields in Salesforce that you plan to sync
            </li>
            <li
              className="css-0"
            >
              System Permission: API Enabled
            </li>
            <li
              className="css-0"
            >
               System Permission: Manage Connected Apps
            </li>
          </ul>
          This is likely a user with a System Administrator or similar profile.
        </div>
        <div
          className="css-old1by"
        >
          <h2
            className="chakra-heading css-1yxyor"
          >
            Create a Salesforce connection using Amazon AppFlow
          </h2>
          <p
            className="chakra-text css-rszk63"
          >
            Amazon AppFlow is a service that allows AWS to connect to Salesforce, for this to work, you need to create a Salesforce connection using Amazon AppFlow.
          </p>
        </div>
        <div
          className="chakra-stack css-6ig1fh"
        >
          <div
            className="css-0"
          >
            <div
              className="css-1ktikgs"
            >
              <img
                alt="AppFlow Connections Page"
                className="chakra-image css-1fyy6mf"
                onClick={[Function]}
                src="appflow-connection-01.png"
              />
            </div>
          </div>
          <div
            className="css-0"
          >
            <ul
              className="css-u3429j"
              role="list"
            >
              <li
                className="css-0"
              >
                Go to the  
                <a
                  className="chakra-link css-oga2ct"
                  href="https://us-east-1.console.aws.amazon.com/appflow/home?region=us-east-1#/connections/salesforce"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  AppFlow connections page 
                  <svg
                    className="chakra-icon css-2y6psj"
                    focusable={false}
                    viewBox="0 0 24 24"
                  >
                    <g
                      fill="none"
                      stroke="currentColor"
                      strokeLinecap="round"
                      strokeWidth="2"
                    >
                      <path
                        d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
                      />
                      <path
                        d="M15 3h6v6"
                      />
                      <path
                        d="M10 14L21 3"
                      />
                    </g>
                  </svg>
                </a>
              </li>
              <li
                className="css-0"
              >
                 Click on the Create connection button 
              </li>
            </ul>
          </div>
        </div>
        <hr
          aria-orientation="horizontal"
          className="chakra-divider css-wcaiv4"
        />
        <div
          className="chakra-stack css-1fsftfy"
        >
          <div
            className="css-0"
          >
            <div
              className="css-1ktikgs"
            >
              <img
                alt="AppFlow Connection Page"
                className="chakra-image css-1fyy6mf"
                onClick={[Function]}
                src="appflow-connection-02.png"
              />
            </div>
          </div>
          <div
            className="css-0"
          >
            <ul
              className="css-u3429j"
              role="list"
            >
              <li
                className="css-0"
              >
                Choose the Salesforce environment you want to connect to
              </li>
              <li
                className="css-0"
              >
                Leave the Private Link selection as 'Disabled'
              </li>
              <li
                className="css-0"
              >
                Leave the Data Encryption setting as is
              </li>
              <li
                className="css-0"
              >
                Type a name for the connection, it can be anything you want, then click Continue
              </li>
            </ul>
          </div>
        </div>
        <hr
          aria-orientation="horizontal"
          className="chakra-divider css-wcaiv4"
        />
        <div
          className="chakra-stack css-1vq7dlv"
        >
          <div
            className="css-0"
          >
            <div
              className="css-1ktikgs"
            >
              <img
                alt="Salesforce Login Page"
                className="chakra-image css-1fyy6mf"
                onClick={[Function]}
                src="appflow-connection-03.png"
              />
            </div>
          </div>
          <div
            className="css-0"
          >
            <ul
              className="css-u3429j"
              role="list"
            >
              <li
                className="css-0"
              >
                Provide the user name and password for the dedicated Salesforce user that you created earlier
              </li>
            </ul>
            <div
              className="css-xbhv25"
            >
              Remember to log in using the newly created dedicated Salesforce user
            </div>
          </div>
        </div>
        <hr
          aria-orientation="horizontal"
          className="chakra-divider css-wcaiv4"
        />
        <div
          className="chakra-stack css-1vq7dlv"
        >
          <div
            className="css-0"
          >
            <div
              className="css-1ktikgs"
            >
              <img
                alt="Salesforce Allow Access Page"
                className="chakra-image css-1fyy6mf"
                onClick={[Function]}
                src="appflow-connection-04.png"
              />
            </div>
          </div>
          <div
            className="css-0"
          >
            <ul
              className="css-u3429j"
              role="list"
            >
              <li
                className="css-0"
              >
                Allow access to your Salesforce data
              </li>
              <li
                className="css-0"
              >
                You will be directed back to the AppFlow connections page, you should see the connection name that you just created. You don't need to do anything else on the AppFlow connections page. Go ahead and select Yes for the next question.
              </li>
            </ul>
          </div>
        </div>
        <div
          className="chakra-stack css-5uflnl"
        >
          <h2
            className="chakra-heading css-wrl52z"
          >
            Did you create a Salesforce connection using Amazon AppFlow?
          </h2>
          <div
            className="css-s2uf1z"
          >
            <button
              className="chakra-button css-1ae6t0z"
              onClick={[Function]}
              type="button"
            >
              Yes
            </button>
          </div>
        </div>
        <div
          className="chakra-collapse"
          style={
            Object {
              "display": "none",
              "height": "0px",
              "opacity": 0,
              "overflow": "hidden",
            }
          }
        >
          <div
            className="css-0"
          />
        </div>
      </div>
      <div
        className="chakra-stack css-1aoc1cb"
      >
        <button
          className="chakra-button css-14g2dfb"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span
            className="chakra-button__icon css-1wh2kri"
          >
            <svg
              aria-hidden={true}
              className="chakra-icon css-onkibi"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                fill="currentColor"
              />
            </svg>
          </span>
          Previous
        </button>
        <button
          className="chakra-button css-jxg557"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          Next
          <span
            className="chakra-button__icon css-1hzyiq5"
          >
            <svg
              aria-hidden={true}
              className="chakra-icon css-onkibi"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <path
                d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                fill="currentColor"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>,
  <span />,
]
`;
